vhdl_syntax/src/parser/productions/interface.rs

```diff
@@ -87,13 +88,114 @@ impl<T: TokenStream> Parser<T> {
             Keyword(Kw::Linkage),
         ]);
     }
+
+    pub fn association_list(&mut self) {
+        self.start_node(AssociationList);
+        self.separated_list(Parser::association_element, Comma);
+        self.end_node();
+    }
+
+    pub fn association_element(&mut self) {
+        self.start_node(AssociationElement);
+
+        let right_arrow_idx = match self.distance_to_closing_paren_or_token(Comma) {
+            Some(length) => self.lookahead_max_distance(length, [RightArrow]),
+            None => {
+                self.eof_err();
```

Comment:
I think that a pure EOF error should at some point be replaced by a more informative error (i.e., unbalanced parentheses, e.t.c.). Also, I think that there should be more stop tokens, such as `semicolon`, `end`, start of declarations and statements, e.t.c. for fault tolerant parsing.
But I'm more or less writing this down as a note so we don't forget it for more error tolerant parsing later. If you don't want to implement this right now, a todo comment or similar would be nice.

Reply:
Tbh. I haven't put too much thought into error handling. This will eventually be an issue with all parser functions anyway, so I figured, going for a half baked solution right now isn't really worth it.

Reply:
It seems like `association_element` has the same problem as `actual_part`: It can only be parsed, if it is followed by a comma or a closing parenthesis. Would you be happy with a similar solution like in `range`? We can look ahead to the end of the `association_list` while parsing `function_call` (or `name` in our case) and pass that information down to `association_element` and `actual_part`.

In that case I would create `*_bounded` functions which take the `length` as a parameter. The public functions would simply pass `usize::MAX` to their `_bounded` variants.